src/server/routes/backend-wallet/import.ts

```diff
@@ -1,39 +1,69 @@
-import { Static, Type } from "@sinclair/typebox";
-import { FastifyInstance } from "fastify";
+import { Type, type Static } from "@sinclair/typebox";
+import type { FastifyInstance } from "fastify";
 import { StatusCodes } from "http-status-codes";
-import { WalletType } from "../../../schema/wallet";
 import { getConfig } from "../../../utils/cache/getConfig";
 import { createCustomError } from "../../middleware/error";
 import { AddressSchema } from "../../schemas/address";
 import { standardResponseSchema } from "../../schemas/sharedApiSchemas";
+import { getGcpKmsResourcePath } from "../../utils/wallets/gcpKmsResourcePath";
 import { importAwsKmsWallet } from "../../utils/wallets/importAwsKmsWallet";
 import { importGcpKmsWallet } from "../../utils/wallets/importGcpKmsWallet";
 import { importLocalWallet } from "../../utils/wallets/importLocalWallet";
 
-const RequestBodySchema = Type.Union([
+const RequestBodySchema = Type.Intersect([
   Type.Object({
-    awsKmsKeyId: Type.String({
-      description: "AWS KMS key ID",
-      examples: ["12345678-1234-1234-1234-123456789012"],
-    }),
-    awsKmsArn: Type.String({
-      description: "AWS KMS key ARN",
-      examples: [
-        "arn:aws:kms:us-east-1:123456789012:key/12345678-1234-1234-1234-123456789012",
-      ],
-    }),
+    label: Type.Optional(
+      Type.String({
+        description: "Optional label for the imported wallet",
+      }),
+    ),
   }),
-  Type.Object({
-    gcpKmsKeyId: Type.String({
-      description: "GCP KMS key ID",
-      examples: ["12345678-1234-1234-1234-123456789012"],
+  Type.Union([
+    Type.Object({
+      awsKmsArn: Type.String({
+        description: "AWS KMS key ARN",
+        examples: [
+          "arn:aws:kms:us-east-1:123456789012:key/12345678-1234-1234-1234-123456789012",
+        ],
+      }),
+      credentials: Type.Object(
```

Comment:
I think there should be a `Type.Optional` here.

Reply:
thanks, good catch